### PR TITLE
CIR-15 Setup CI

### PR DESCRIPTION
**Pull request by reach4nk**
Refer to https://pre-commit.com/#install to install precommit hook on your local dev environment
* Added .pre-commit-config.yaml which automatically fixes many formatting issues, and runs checks like autopep8, mypy
* This auto fixed many files' formatting, that is the cause of 20+ files changes
* And had to fix few files, as they were showing error code on mypy.

### CODE REVIEW

**Review by tidmanmike — APPROVED** [review]

File: config/__init__.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (mateen-hussain) [inline config/__init__.py:1]: why do we need this import ? Where are we using it ?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    -   id: check-yaml
+    -   id: debug-statements
+    -   id: double-quote-string-fixer
+    -   id: name-tests-test
+    -   id: requirements-txt-fixer
+-   repo: https://github.com/asottile/reorder_python_imports
+    rev: v3.9.0
+    hooks:
+    -   id: reorder-python-imports
+        args: [--py37-plus, --add-import, 'from __future__ import annotations']
```
Comment (mateen-hussain) [inline .pre-commit-config.yaml:16]: I havent found a good reason to import this `from __future__ import annotations` in all our files. And also this is future stuff, which means might not be supported widely and not tested also. 

Unless there is a good use of it, let's remove this and all imports in the code.

File: config/__init__.py (hunk 1 above)
Comment (reach4nk) [reply config/__init__.py]: Removed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (reach4nk) [reply .pre-commit-config.yaml]: Totally agree, removed.

**Review by mateen-hussain — APPROVED** [review]

**Review by tidmanmike — APPROVED** [review]

**Pull request merged**